src/lib/Toast.svelte

```diff
@@ -109,14 +108,33 @@
 
 	let timeoutId: ReturnType<typeof setTimeout>;
 
+	let remainingTime = toast.duration || duration || TOAST_LIFETIME;
+
+	let toastUpdateCount = 0;
+
+	$: if (toast) {
+		toastUpdateCount++;
+	}
+
+	$: if (toastUpdateCount > 1 && timeoutId) {
+		// if the toast has been updated after the initial render,
+		// we want to reset the timer and set the remaining time to the
+		// new duration
+		clearTimeout(timeoutId);
+		remainingTime = toast.duration || duration || TOAST_LIFETIME;
+		startTimer();
```

Comment:
Can confirm that these lines are causing all toasts to dismiss at the same time https://github.com/wobsoriano/svelte-sonner/issues/54